src/thriveopedia/Thriveopedia.cs

```diff
@@ -0,0 +1,338 @@
+
+    /// <summary>
+    ///   Page history for backwards navigation.
+    /// </summary>
+    private Stack<ThriveopediaPage> pageHistory = new();
+
+    /// <summary>
+    ///   Page future for forwards navigation (if the player has already gone backwards).
+    /// </summary>
+    private Stack<ThriveopediaPage> pageFuture = new();
+
+    [Signal]
+    public delegate void OnThriveopediaClosed();
+
+    /// <summary>
+    ///   The currently open Thriveopedia page. Defaults to the home page if none has been set.
+    /// </summary>
+    public ThriveopediaPage SelectedPage
+    {
+        get => selectedPage ?? homePage;
+        set
+        {
+            // Hide the last page and show the new page
+            selectedPage?.Hide();
+
+            selectedPage = value;
+            selectedPage.Show();
+
+            // The home page is always the first in the history, so ignore it
+            backButton.Disabled = pageHistory.Count <= 1;
+            forwardButton.Disabled = pageFuture.Count == 0;
+
+            pageTitle.Text = SelectedPage.TranslatedPageName;
+            SelectInTreeWithoutEvent(selectedPage);
+        }
+    }
+
+    /// <summary>
+    ///   Details for the game currently in progress. Null if opened from the main menu. When set, initialises all
+    ///   pages requiring an active game.
+    /// </summary>
+    public GameProperties? CurrentGame
+    {
+        get => currentGame;
+        set
+        {
+            // We only want to initialise these pages once per game
+            if (value == currentGame)
+                return;
+
+            currentGame = value;
+
+            // Add all pages associated with a game in progress
+            if (currentGame != null)
+            {
+                AddPage("CurrentWorld");
+                AddPage("PatchMap", "CurrentWorld");
+                AddPage("EvolutionaryTree", "CurrentWorld");
+            }
+
+            // Notify all pages of the new game properties
+            foreach (var page in allPages.Keys)
+                page.CurrentGame = currentGame;
+        }
+    }
+
+    public override void _Ready()
+    {
+        backButton = GetNode<TextureButton>(BackButtonPath);
+        forwardButton = GetNode<TextureButton>(ForwardButtonPath);
+        pageContainer = GetNode<MarginContainer>(PageContainerPath);
+        pageTreeContainer = GetNode<PanelContainer>(PageTreeContainerPath);
+        pageTitle = GetNode<Label>(PageTitlePath);
+        pageTree = GetNode<Tree>(PageTreePath);
+
+        // Create and hide a blank root to avoid home being used as the root
+        pageTree.CreateItem();
+        pageTree.HideRoot = true;
+
+        // Keep a special reference to the home page
+        homePage = GetNode<ThriveopediaHomePage>(HomePagePath);
+        allPages.Add(homePage, CreateTreeItem(homePage, null));
+
+        // Add all pages not associated with a game in progress
+        AddPage("Museum");
```

Comment:
I assume this doesn't take too much time because otherwise we should think about only initializing the Thriveopedia when it becomes visible.

Reply:
Right now it doesn't, since for the museum in particular the cards are only created when it becomes visible.

It may take longer if the list of pages grows a lot bigger.